packages/aws-rfdk/lib/deadline/test/repository.test.ts

```diff
@@ -1224,3 +1225,45 @@ test('imports repository settings', () => {
   const installerGroup = repository.node.tryFindChild('Installer') as AutoScalingGroup;
   expect(installerGroup.userData.render()).toContain(`aws s3 cp '${repositorySettings.s3ObjectUrl}'`);
 });
+
+test('secret manager enabled', () => {
+  // GIVEN
+  const expectedCredentials = new Secret(stack, 'CustomSMAdminUser', {
+    description: 'Custom admin credentials for the Secret Management',
+    generateSecretString: {
+      excludeCharacters: '\"$&\'()-/<>[\\]\`{|}',
+      includeSpace: false,
+      passwordLength: 24,
+      requireEachIncludedType: true,
+      generateStringKey: 'password',
+      secretStringTemplate: JSON.stringify({ username: 'admin' }),
+    },
+  });
+
+  // WHEN
+  const repository = new Repository(stack, 'Repository', {
+    vpc,
+    version,
+    secretsManagementSettings: {
+      enabled: true,
+      credentials: expectedCredentials,
+    },
+  });
+
+  // THEN
+  expect(repository.secretsManagementSettings.credentials).toBe(expectedCredentials);
```

Comment:
We should have another expectation on the User Data to make sure it includes the line to retrieve the credentials. Something like:
```
const installerGroup = repository.node.tryFindChild('Installer') as AutoScalingGroup;
expect(installerGroup.userData.render())
  .toContain(`aws secretsmanager get-secret-value --secret-id=${repository.secretsManagementSettings.credentials.secretArn}'`);
```

Reply:
I believe this script is running not in UserData and I do not know a way to check script that is running on the instance.

Reply:
I'm pretty sure it does get added to the User Data of the repository installer instance. The code you added went into `installDeadlineRepository.sh`. You can see the asset get created as a `ScriptAsset` [here](https://github.com/aws/aws-rfdk/blob/mainline/packages/aws-rfdk/lib/deadline/lib/repository.ts#L917) and then added to the User Data [here](https://github.com/aws/aws-rfdk/blob/mainline/packages/aws-rfdk/lib/deadline/lib/repository.ts#L951) using the [executeOn](https://github.com/aws/aws-rfdk/blob/984201ca356c83c8e476ec5028c252f772c442bd/packages/aws-rfdk/lib/core/lib/script-assets.ts#L178) method from out `ScriptAsset` class.

Reply:
I think we might be a bit confused about what's going on here. AFAIK, the changes to use the AWS CLI to grab the secret from SM happens in the `installDeadlineRepository.sh` script. The UserData downloads, unzips, and runs that script with the parameters we provide, but it does so without knowing the contents of the script itself, so I don't think we can actually assert the AWS CLI call here.

However, I think we _can_ assert that the new SM-related parameters are being passed to the `installDeadlineRepository.sh` script, which would be (roughly) in the following format in UserData:
```
<path-to-installDeadlineRepository.sh> ... <sm-enabled> <region> <sm-admin-creds-secret-arn> ...
```

Note that the above will change once this PR is rebased since the repo installer script uses named parameters instead of positional parameters